runtime/model-json/src/main/java/io/aklivity/zilla/runtime/model/json/internal/JsonModelHandler.java

```diff
@@ -83,8 +96,47 @@ protected final boolean validate(
             status &= provider != null;
             if (status)
             {
+                for (OctetsFW value: extracted.values())
+                {
+                    value.wrap(EMPTY_BUFFER, 0, 0);
+                }
                 in.wrap(buffer, index, length);
-                provider.createReader(in).readValue();
+                parser = provider.createParser(in);
+                OctetsFW valueBytes = null;
+                while (parser.hasNext())
+                {
+                    JsonParser.Event event = parser.next();
+                    if (!extracted.isEmpty())
+                    {
+                        switch (event)
+                        {
+                        case KEY_NAME:
+                            String key = parser.getString();
+                            valueBytes = extracted.get(key);
+                            break;
+                        case VALUE_STRING:
+                            if (valueBytes != null)
+                            {
+                                int offset = (int) parser.getLocation().getStreamOffset() - DOUBLE_QUOTE_LENGTH;
+                                offset += index;
+                                int valLength = parser.getString().getBytes(UTF_8).length;
+                                valueBytes.wrap(in.buffer(), offset - valLength, offset);
+                                valueBytes = null;
+                            }
+                            break;
+                        case VALUE_NUMBER:
+                            if (valueBytes != null)
+                            {
+                                int offset = (int) parser.getLocation().getStreamOffset();
+                                offset += index;
+                                int valLength = parser.getString().getBytes().length;
```

Comment:
We should be able to calculate the UTF-8 bytes length of this String without allocating the bytes themselves.